<launch>

  <arg name="HAND" default="left" />
  <arg name="NUM" default="0" />
  <arg name="CONTROLLER" default="torque" />
  <arg name="RESPAWN" default="true" />
  <arg name="VISUALIZE" default="true" />
  <!-- <arg name="ZEROS" default="$(find allegro_hand_parameters)/zero_files/zero_SAH020BL016.yaml" /> -->
  
  <arg name="ZEROS" default="$(find allegro_hand_parameters)/zero.yaml" />

  <arg name="AUTO_CAN" default="true" />  <!--If true, ignores CAN_DEVICE. -->
  <arg name="CAN_DEVICE" default="/dev/pcanusb1" />
  <arg name="JSP_GUI" default="false" />
  <arg name="KEYBOARD" default="false" />
  <arg name="GAZEBO" default="false" />  <!--use at your own risk-->

  <include file="$(find allegro_hand_controllers)/launch/allegro_hand.launch">
    <arg name="HAND" value="$(arg HAND)"/>
    <arg name="NUM" value="$(arg NUM)"/>
    <arg name="CONTROLLER" value="$(arg CONTROLLER)"/>
    <arg name="RESPAWN" value="$(arg RESPAWN)"/>
    <arg name="VISUALIZE" value="$(arg VISUALIZE)"/>
    <arg name="ZEROS" value="$(arg ZEROS)"/>
    <arg name="AUTO_CAN" value="$(arg AUTO_CAN)"/>
    <arg name="CAN_DEVICE" value="$(arg CAN_DEVICE)"/>
    <arg name="JSP_GUI" value="$(arg JSP_GUI)"/>
    <arg name="KEYBOARD" value="$(arg KEYBOARD)"/>
    <arg name="GAZEBO" value="$(arg GAZEBO)"/>
  </include>

  <include file="$(find allegro_toolkit)/launch/vrpn_optitrack.launch">
  </include>

  <node name="ad_control" pkg="allegro_toolkit" type="ad_control" output="screen">
  </node>

  <node name="plotter"
        pkg="rqt_plot"
        type="rqt_plot"
        args="/hand/plotvar/data[0] /hand/plotvar/data[1] /hand/plotvar/data[2] /hand/plotvar/data[3]" >
  </node>
  <!-- <node name="bio_sensors" pkg="biotac_sensors" type="biotac_pub">
  </node> -->

</launch>